client/src/common/components/Screen/Screen.tsx

```diff
@@ -13,64 +17,73 @@ const Wrapper = styled.View<{backgroundColor?: string}>(
   }),
 );
 
-const TopBarWrapper = styled.View({
-  zIndex: 100,
-  flexDirection: 'row',
-  alignItems: 'center',
-  paddingLeft: SPACINGS.SIXTEEN,
-  paddingRight: SPACINGS.SIXTEEN,
-  height: SPACINGS.FOURTYFOUR,
-  width: '100%',
-  justifyContent: 'space-between',
-});
-
-const BackButton = styled(IconButton)({
-  marginLeft: -SPACINGS.SIXTEEN,
-});
+type TopBarWrapperProps = ViewProps & {
+  floating?: boolean;
+  safeArea: EdgeInsets;
+};
 
-const TopBar: React.FC<{
-  hasDarkBackground?: boolean;
-  onPressBack?: () => void;
-}> = ({onPressBack, hasDarkBackground: isOnDarkBackground}) => (
-  <>
-    <TopSafeArea minSize={SPACINGS.SIXTEEN} />
-    <TopBarWrapper>
-      {onPressBack && (
-        <BackButton
-          noBackground
-          onPress={onPressBack}
-          Icon={ArrowLeftIcon}
-          fill={isOnDarkBackground ? COLORS.WHITE : COLORS.BLACK}
-        />
-      )}
-    </TopBarWrapper>
-  </>
+const TopBarWrapper = styled.View<TopBarWrapperProps>(
+  ({floating, safeArea}) => ({
+    zIndex: 100,
+    flexDirection: 'row',
+    alignItems: 'center',
+    paddingTop: safeArea.top,
+    height: safeArea.top + SPACINGS.FOURTYFOUR,
+    width: '100%',
+    justifyContent: 'space-between',
+    position: floating ? 'absolute' : 'relative',
+  }),
 );
 
+type OverlayProps = LinearGradientProps & {safeArea: EdgeInsets};
+
+const Overlay = styled(LinearGradient)<OverlayProps>(({safeArea}) => ({
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  right: 0,
+  height: safeArea.top + SPACINGS.FOURTYFOUR,
+}));
+
 type ScreenProps = {
   backgroundColor?: string;
+  textColor?: string;
   children: React.ReactNode;
-  hasDarkBackground?: boolean;
-  noStatusBar?: boolean;
-  noTopBar?: boolean;
   onPressBack?: () => void;
-  title?: string;
+  floatingTopBar?: boolean;
+  topBarOverlay?: boolean;
 };
 
 const Screen: React.FC<ScreenProps> = ({
   backgroundColor,
+  textColor,
   children,
-  hasDarkBackground,
-  noTopBar,
   onPressBack,
+  floatingTopBar = true,
+  topBarOverlay = false,
 }) => {
+  const safeArea = useSafeAreaInsets();
+
   return (
     <Wrapper backgroundColor={backgroundColor}>
-      {!noTopBar && (
-        <TopBar
-          onPressBack={onPressBack}
-          hasDarkBackground={hasDarkBackground}
-        />
+      {onPressBack && (
+        <TopBarWrapper floating={floatingTopBar} safeArea={safeArea}>
+          {topBarOverlay && (
+            <Overlay
+              colors={[COLORS.WHITE, COLORS.WHITE_TRANSPARENT]}
```

Comment:
Should'nt this be based on the `backgroundColor` instead?

Reply:
I tried but couldn't get it to work - wanna help?

Reply:
Ooooh I know now why it faild - I used 'red instead of hex. I'll fix that and push - great that you pointed it uot!